examples/platforms/gp712/radio.c

```diff
@@ -0,0 +1,309 @@
+    QORVO_LQI_BIT_MASK = 0x7f,
+};
+
+extern otRadioFrame  sTransmitFrame;
+
+static otRadioState  sState;
+static otInstance   *pQorvoInstance;
+
+typedef struct otCachedSettings_s {
+    uint16_t panid;
+} otCachedSettings_t;
+static otCachedSettings_t otCachedSettings;
+
+static uint8_t  sScanstate          = 0;
+static int8_t   sLastReceivedPower  = 127;
+
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    (void)aInstance;
+    qorvoRadioGetIeeeEui64(aIeeeEui64);
+}
+
+void otPlatRadioSetPanId(otInstance *aInstance, uint16_t panid)
+{
+    (void)aInstance;
+    qorvoRadioSetPanId(panid);
+    otCachedSettings.panid = panid;
+}
+
+void otPlatRadioSetExtendedAddress(otInstance *aInstance, uint8_t *address)
+{
+    (void)aInstance;
+    qorvoRadioSetExtendedAddress(address);
+}
+
+void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t address)
+{
+    (void)aInstance;
+    qorvoRadioSetShortAddress(address);
+}
+
+bool otPlatRadioIsEnabled(otInstance *aInstance)
+{
+    (void)aInstance;
+    return (sState != OT_RADIO_STATE_DISABLED);
+}
+
+otError otPlatRadioEnable(otInstance *aInstance)
+{
+    pQorvoInstance = aInstance;
+    memset(&otCachedSettings, 0x00, sizeof(otCachedSettings_t));
+
+    if (!otPlatRadioIsEnabled(aInstance))
+    {
+        sState = OT_RADIO_STATE_SLEEP;
+    }
+
+    return OT_ERROR_NONE;
+}
+
+otError otPlatRadioDisable(otInstance *aInstance)
+{
+    (void)aInstance;
+    otEXPECT(otPlatRadioIsEnabled(aInstance));
+
+    if (sState == OT_RADIO_STATE_RECEIVE)
+    {
+        qorvoRadioSetRxOnWhenIdle(false);
+    }
+    sState = OT_RADIO_STATE_DISABLED;
+
+exit:
+    return OT_ERROR_NONE;
+}
+
+otError otPlatRadioSleep(otInstance *aInstance)
+{
+    (void)aInstance;
+    otError error = OT_ERROR_INVALID_STATE;
+
+    if (sState == OT_RADIO_STATE_RECEIVE)
+    {
+        qorvoRadioSetRxOnWhenIdle(false);
+        error = OT_ERROR_NONE;
+        sState = OT_RADIO_STATE_SLEEP;
+    }
+
+    return error;
+}
+
+otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
+{
+    otError error = OT_ERROR_INVALID_STATE;
+    pQorvoInstance = aInstance;
+
+    if ((sState != OT_RADIO_STATE_DISABLED) && (sScanstate==0))
+    {
+        qorvoRadioSetCurrentChannel(aChannel);
+        error = OT_ERROR_NONE;
+    }
+    if (sState == OT_RADIO_STATE_SLEEP)
+    {
+        qorvoRadioSetRxOnWhenIdle(true);
+        error = OT_ERROR_NONE;
+        sState = OT_RADIO_STATE_RECEIVE;
+    }
+
+    return error;
+
+}
+
+otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aPacket)
+{
+    otError err = OT_ERROR_NONE;
+
+    pQorvoInstance = aInstance;
+
+    otEXPECT_ACTION(sState != OT_RADIO_STATE_DISABLED, err = OT_ERROR_INVALID_STATE);
+
+    err = qorvoRadioTransmit(aPacket);
+
+exit:
+    return err;
+}
+
+void cbQorvoRadioTransmitDone(otRadioFrame *aPacket, bool aFramePending, otError aError)
+{
+    otPlatRadioTransmitDone(pQorvoInstance, aPacket, aFramePending, aError);
```

Comment:
I understand the remark, but this is not a trivial change and we do not plan to do that immediately.